### PR TITLE
PIL-674: Ethers v5 migration

### PR DESCRIPTION
**Pull request by pedrouid**
This PR migrates `ethers` to v5 prior to Connext integration. Missing tasks:
- [x] Replace `scrypt` shim with native bindings
- [x] Replace `wallet.encrypt` method with native bindings

I think both of these native bindings can be done externally to ethers without requiring to fork (especially now that ethers v5 doesn't polyfill getRandomValues internally)

cc @poocart

### CODE REVIEW

**Comment by poocart** [comment]
Related Linear ticket – https://linear.app/pillarproject/issue/PIL-674/update-to-ethers-v5.

File: patches/ethers+5.0.0-beta.190.patch [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+-	                        }
+-	                        resolve(new Uint8Array(key));
+-	                    } else if (progressCallback && progress !== lastProgress) {
+-	                        lastProgress = progress;
+-	                        return progressCallback(progress);
+-	                    }
+-	                });
+-	            });
++	        	return new Promise(function(resolve, reject) {
++								let lastProgress = 0;
```
Comment (jsidorenko) [inline patches/ethers+5.0.0-beta.190.patch:32]: the indentation looks wrong

Comment (poocart) [reply patches/ethers+5.0.0-beta.190.patch:32]: agree, but I think patcher recognizes it wrong, didn't put too much effort in trying to fix this, but will check once again

Comment (jsidorenko) [reply patches/ethers+5.0.0-beta.190.patch:32]: looks like it was replaced with tabs instead of spaces

Comment (poocart) [reply patches/ethers+5.0.0-beta.190.patch]: was a tricky one, starts with tab, ends with spacing :) copying over exact indentation worked, thanks 👍

**Pull request merged**